src/transformers/models/roformer/tokenization_roformer.py

```diff
@@ -0,0 +1,150 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for RoFormer."""
+
+from ...utils import logging
+from ..bert.tokenization_bert import BasicTokenizer, BertTokenizer
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "junnyu/roformer_chinese_small": "https://huggingface.co/junnyu/roformer_chinese_small/resolve/main/vocab.txt",
+        "junnyu/roformer_chinese_base": "https://huggingface.co/junnyu/roformer_chinese_base/resolve/main/vocab.txt",
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {"junnyu/roformer_chinese_small": 512, "junnyu/roformer_chinese_base": 512}
+
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "junnyu/roformer_chinese_small": {"do_lower_case": True},
+    "junnyu/roformer_chinese_base": {"do_lower_case": True},
+}
+
+
+class CustomBasicTokenizer(BasicTokenizer):
+    def __init__(self, vocab, do_lower_case=True, never_split=None, tokenize_chinese_chars=True, strip_accents=None):
+        super().__init__(
+            do_lower_case=do_lower_case,
+            never_split=never_split,
+            tokenize_chinese_chars=tokenize_chinese_chars,
+            strip_accents=strip_accents,
+        )
+
+        self.vocab = vocab
+
+    def _tokenize_chinese_chars(self, text):
+        try:
+            import jieba
+        except ImportError:
+            logger.warning(
+                "Can't import jieba, it will tokenize chinese chars like BertTokenizer. If you want to use jieba, please pip install jieba first."
+            )
+            output = []
+            for char in text:
+                cp = ord(char)
+                if self._is_chinese_char(cp):
+                    output.append(" ")
+                    output.append(char)
+                    output.append(" ")
+                else:
+                    output.append(char)
+            return "".join(output)
+
+        output = []
+        for wholeword in jieba.cut(text, HMM=False):
+            if wholeword in self.vocab:
+                output.append(" ")
+                output.append(wholeword)
+                output.append(" ")
+            else:
+                for char in wholeword:
+                    cp = ord(char)
+                    if self._is_chinese_char(cp):
+                        output.append(" ")
+                        output.append(char)
+                        output.append(" ")
+                    else:
+                        output.append(char)
+        return "".join(output)
+
+
+class RoFormerTokenizer(BertTokenizer):
```

Comment:
Could we try to not inherit from `BertTokenizer` & instead just copy & past all the functionality in here?

Reply:
+1